invisible_cities/cities/components_test.py

```diff
@@ -568,8 +569,13 @@ def test_write_city_configuration(config_tmpdir):
 
     df = pd.read_hdf(filename, "/config/" + city_name).set_index("variable")
     for var, value in args.items():
-        assert var in df.index
-        assert str(value) == df.value.loc[var]
+        if not isinstance(value, dict):
+            assert var in df.index
+            assert str(value) == df.value.loc[var]
+        else:
+            for var1, value1 in value.items():
+                assert f"{var}.{var1}" in df.index        
+                assert  str(value1) == df.value.loc[f"{var}.{var1}"]
```

Comment:
I think in this particular case it will be more readable to ignore the entry `"g"` in the loop and run the assertion separately after the loop, especially if you implement another level of nesting as suggested above.